libs/core/langchain_core/runnables/fallbacks.py

```diff
@@ -171,6 +186,10 @@ async def ainvoke(
         config: Optional[RunnableConfig] = None,
         **kwargs: Optional[Any],
     ) -> Output:
+        if self.exception_key is not None and not isinstance(input, dict):
+            raise ValueError(
+                "If 'exception_key' is specified then input must be a dictionary."
```

Comment:
Could you expand the exception message: 

"However found a type of `{type(input)}` for input" 

or something to that extent to make the error message more detailed